apps/editing-toolkit/editing-toolkit-plugin/wpcom-site-editor/navigation-toggle/custom-navigation-toggle-button.js

```diff
@@ -0,0 +1,41 @@
+import { Button } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { Icon, wordpress } from '@wordpress/icons';
+
+export default function CustomNavigationToggleButton() {
+	const siteIconUrl = useSelect( ( select ) => {
+		const { getEntityRecord } = select( 'core' );
+		return getEntityRecord( 'root', '__unstableBase', undefined )?.site_icon_url;
+	} );
```

Comment:
This all seems to work well, but I wonder if we can avoid having to build and inject the entire new component?

Cant we just select `.edit-site-navigation-toggle` and add a 'click' event listener that does the same as the onClick in this component?  I tried this in the console and it seemed to work as well:

`document.querySelector( '.edit-site-navigation-toggle' ).addEventListener( 'click', ( ev ) => { ev.preventDefault(); ev.stopPropagation(); window.location.href = './index.php' } )`

Reply:
Done, thanks for the suggestion!
MutationObserver + addEventListener = PROFIT!